### PR TITLE
HLTDQM : fix memory issue (part 1)

### PR DESCRIPTION
**Pull request by mtosi**
as reported in [1] and discussed during (and after) the meeting
there is the need of squeezing as much as possible the memory used by the HLTDQM

this is the 1st --low level-- round, in which
1. I disable the tracking cumulative hit efficiency vs PU, lumi and BX almost everywhere
2. reduce the #bins in the JetMET directories, by updating [2]
   * disabling some not really useful plots
      (like the 2D in the HEM and HEP slices, which are simply a zoom of the full eta-phi map)
   * reducing the bin width in the eta plot [from 50 to 20 bins (from 0.2 to 0.5 resolution) ]
      the best would be to move to variable binning
   * dropping plots vs LS done per partitions

in this occasion, I also clean / re-organize the directories structure
==> *the comparison plot by plot will fail*
this is probably not finished yet :(

@fwyzard @davidlange6 

[1]
in https://indico.cern.ch/event/658725/contributions/2685778/attachments/1505407/2345690/dqm_memory.pdf

[2]
https://github.com/cms-sw/cmssw/commit/c74aaa72944796340389d7006a5b6f0cfc6ff185#diff-c69d426b0a077e41f02e7acd0f66d649

### CODE REVIEW

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by davidlange6** [comment]
please test

**Comment by mtosi** [comment]
btw, 
after having followed the suggestion triggered by the `scram build code-checks`,
I ran the command, and I would suggest to add at the end either a file or a message
which clearly states the outcome

**Comment by fwyzard** [comment]
> btw, after having followed the suggestion triggered by the scram build
code-checks, I ran the command, and I would suggest to add at the end
either a file or a message which clearly states the outcome

What do you mean ?
What did you run, and what would you like to see ?

.A

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by mtosi** [comment]
waiting for instructions on how to produce the pie chart [1]

I can say that the DQM file I got decreases in terms of size
workflow 136.731 : 91M --> 87M 
workflow 10042.0 : 123M --> 119M
=> in both cases I see a gain of 4 MB, which is few %

[1]
https://micius.web.cern.ch/micius/code40_profile/mbGraph.html?#?profile=10042_step3.json&show=profile

**Comment by mtosi** [comment]
running a quick-n-dirt script which checks the #bins 
I found the following #bins 

HLT 96576580

HIG 104688 ( 0.1 %)
BPH 10624 ( 0.0 %)
EventInfo 96 ( 0.0 %)
LumiMonitoring 2502 ( 0.0 %)
TopHLTOffline 790149 ( 0.8 %)
MET 35148 ( 0.0 %)
EXO 1989114 ( 2.1 %)
Higgs 780481 ( 0.8 %)
B2G 494848 ( 0.5 %)
HCAL 240343 ( 0.2 %)
JME 5473790 ( 5.7 %)
GeneralHLTOffline 16 ( 0.0 %)
SUSY 415076 ( 0.4 %)
FSQ 28386 ( 0.0 %)
Objects 91 ( 0.0 %)
Tracking 778016 ( 0.8 %)
BTV 963551 ( 1.0 %)
SiStrip 1560553 ( 1.6 %)
EGM 810699 ( 0.8 %)
Pixel 81744493 ( 84.6 %)
TAU 50160 ( 0.1 %)
Vertexing 202830 ( 0.2 %)
HT 37506 ( 0.0 %)
HI 63420 ( 0.1 %)

=> in this version, the most offending directory is the Pixel !
because of 4 histos
DQM_V0001_R000274199__Global__CMSSW_X_Y_Z__RECO.root:/DQMData/Run 274199/HLT/Run summary/Pixel/TrackClusters/num_clusters_ontrack_per_instLumi_PXBarrel 20018004 ( 24.5 %)
DQM_V0001_R000274199__Global__CMSSW_X_Y_Z__RECO.root:/DQMData/Run 274199/HLT/Run summary/Pixel/num_clusters_per_instLumi_PXForward 20018004 ( 24.5 %)
DQM_V0001_R000274199__Global__CMSSW_X_Y_Z__RECO.root:/DQMData/Run 274199/HLT/Run summary/Pixel/num_clusters_per_instLumi_PXBarrel 20018004 ( 24.5 %)
DQM_V0001_R000274199__Global__CMSSW_X_Y_Z__RECO.root:/DQMData/Run 274199/HLT/Run summary/Pixel/TrackClusters/num_clusters_ontrack_per_instLumi_PXForward 20018004 ( 24.5 %)
!!!

dropping 2 of them,
and slightly reducing their binning, I get
TOT 60880337

**HLT 18114172 ( 29.8 %)**

BPH 10624 ( 0.1 %)
GeneralHLTOffline 16 ( 0.0 %)
FSQ 28386 ( 0.2 %)
HI 63420 ( 0.4 %)
EGM 810699 ( 4.5 %)
B2G 494848 ( 2.7 %)
**Pixel 3282085 ( 18.1 %)**
TAU 50160 ( 0.3 %)
**SiStrip 1560553 ( 8.6 %)**
Objects 91 ( 0.0 %)
HIG 104688 ( 0.6 %)
HCAL 240343 ( 1.3 %)
**JME 5473790 ( 30.2 %)**
Tracking 778016 ( 4.3 %)
Higgs 780481 ( 4.3 %)
TopHLTOffline 790149 ( 4.4 %)
LumiMonitoring 2502 ( 0.0 %)
HT 37506 ( 0.2 %)
SUSY 415076 ( 2.3 %)
BTV 963551 ( 5.3 %)
EventInfo 96 ( 0.0 %)
Vertexing 202830 ( 1.1 %)
MET 35148 ( 0.2 %)
**EXO 1989114 ( 11.0 %)**

**Comment by mtosi** [comment]
plain 94X
TOT 149965200
HLT 107199035 ( 71.5 %)
BPH 10624 ( 0.0 %)
GeneralHLTOffline 16 ( 0.0 %)
NoBPTX 114512 ( 0.1 %)
FSQ 28386 ( 0.0 %)
HI 63420 ( 0.1 %)
Pixel 81744493 ( 76.3 %)
LepHT 5196 ( 0.0 %)
SiStrip 1560553 ( 1.5 %)
Photon 306916 ( 0.3 %)
Objects 91 ( 0.0 %)
EXO 3421524 ( 3.2 %)
HCAL 240343 ( 0.2 %)
EgOffline 435 ( 0.0 %)
B2GMonitor 975160 ( 0.9 %)
Tracking 1723200 ( 1.6 %)
Higgs 780481 ( 0.7 %)
TopHLTOffline 790149 ( 0.7 %)
EG 1298352 ( 1.2 %)
LumiMonitoring 2502 ( 0.0 %)
Exotica 182402 ( 0.2 %)
HT 205634 ( 0.2 %)
SUSY 178970 ( 0.2 %)
TauOffline 50160 ( 0.0 %)
JetMET 11407162 ( 10.6 %)
EGTagAndProbeEffs 270984 ( 0.3 %)
BTV 963551 ( 0.9 %)
EventInfo 96 ( 0.0 %)
Vertexing 202830 ( 0.2 %)
MET 438663 ( 0.4 %)
Hotline 1320 ( 0.0 %)
SusyHLTOffline 230910 ( 0.2 %)

current version of this PR
TOT 60880337
HLT 18114172 ( 29.8 %)
BPH 10624 ( 0.1 %)
GeneralHLTOffline 16 ( 0.0 %)
FSQ 28386 ( 0.2 %)
HI 63420 ( 0.4 %)
EGM 810699 ( 4.5 %)
B2G 494848 ( 2.7 %)
Pixel 3282085 ( 18.1 %)
TAU 50160 ( 0.3 %)
SiStrip 1560553 ( 8.6 %)
Objects 91 ( 0.0 %)
HIG 104688 ( 0.6 %)
HCAL 240343 ( 1.3 %)
JME 5473790 ( 30.2 %)
Tracking 778016 ( 4.3 %)
Higgs 780481 ( 4.3 %)
TopHLTOffline 790149 ( 4.4 %)
LumiMonitoring 2502 ( 0.0 %)
HT 37506 ( 0.2 %)
SUSY 415076 ( 2.3 %)
BTV 963551 ( 5.3 %)
EventInfo 96 ( 0.0 %)
Vertexing 202830 ( 1.1 %)
MET 35148 ( 0.2 %)
EXO 1989114 ( 11.0 %)

**Comment by davidlange6** [comment]
please test

**Comment by mtosi** [comment]
I have just checked the #histogram per directory [1]
and it seems that --on average-- there are no real outliers, now

JME directory should be reduced even further ...
while, Strip, Pixel and Tracking should be enable only on HLTMonitor PD

[1]
TOT 60880337 #histos: 104186
HLT 18114172 ( 29.8 %) #histos: 29915 ( 28.7 %)
BPH 10624 ( 0.1 %), #histos: 84 ( 0.3 %)
B2G 494848 ( 2.7 %), #histos: 504 ( 1.7 %)
HCAL 240343 ( 1.3 %), #histos: 117 ( 0.4 %)
Higgs 780481 ( 4.3 %), #histos: 2117 ( 7.1 %)
LumiMonitoring 2502 ( 0.0 %), #histos: 1 ( 0.0 %)
JME 5473790 ( 30.2 %), #histos: 7188 ( 24.0 %)
FSQ 28386 ( 0.2 %), #histos: 419 ( 1.4 %)
Pixel 3282085 ( 18.1 %), #histos: 131 ( 0.4 %)
EXO 1989114 ( 11.0 %), #histos: 2574 ( 8.6 %)
TAU 50160 ( 0.3 %), #histos: 475 ( 1.6 %)
HT 37506 ( 0.2 %), #histos: 56 ( 0.2 %)
HI 63420 ( 0.4 %), #histos: 393 ( 1.3 %)
SUSY 415076 ( 2.3 %), #histos: 1914 ( 6.4 %)
EGM 810699 ( 4.5 %), #histos: 3409 ( 11.4 %)
MET 35148 ( 0.2 %), #histos: 72 ( 0.2 %)
GeneralHLTOffline 16 ( 0.0 %), #histos: 3 ( 0.0 %)
EventInfo 96 ( 0.0 %), #histos: 4 ( 0.0 %)
Objects 91 ( 0.0 %), #histos: 4 ( 0.0 %)
Tracking 778016 ( 4.3 %), #histos: 1086 ( 3.6 %)
TopHLTOffline 790149 ( 4.4 %), #histos: 4492 ( 15.0 %)
Vertexing 202830 ( 1.1 %), #histos: 207 ( 0.7 %)
BTV 963551 ( 5.3 %), #histos: 3408 ( 11.4 %)
SiStrip 1560553 ( 8.6 %), #histos: 1077 ( 3.6 %)
HIG 104688 ( 0.6 %), #histos: 180 ( 0.6 %)

**Comment by dmitrijus** [comment]
+1

**Comment by mtosi** [comment]
thanks !!!
may I ask if a backport for 92x will allow us to start integrating new PRs related to the HLTDQM ?
@davidlange6 @slava77 @dmitrijus @fwyzard

File: DQMOffline/Trigger/plugins/JetMonitor.cc [hunk 1]
```diff
@@ -105,7 +105,8 @@ void JetMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
   std::string hist_obtag = "";
   std::string histtitle_obtag = "";
   std::string currentFolder = folderName_ ;
-  ibooker.setCurrentFolder(currentFolder.c_str());
+  //  ibooker.setCurrentFolder(currentFolder.c_str());
```
Comment (davidlange6) [inline DQMOffline/Trigger/plugins/JetMonitor.cc]: hi @mtosi - go ahead and remove the commented line

**Comment by mtosi** [comment]
hltdqm

**Comment by mtosi** [comment]
I'm going to commit the requested code updates

this is the further decrease in memory

hltdqm

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by mtosi** [comment]
@dmitrijus @davidlange6 
I think I'm done w/ this
(sorry for having added extra commits in the last round, but they should simply decrease even more the size)

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by davidlange6** [comment]
hi @mtosi - please rebase this to remove things already merged in from the commit history. Thanks

**Comment by mtosi** [comment]
i will do it tomorrow
  mia

**Comment by mtosi** [comment]
@davidlange6 
I've just rebased w.r.t. 94x
is it ok, now ?
I do not see any conflicts w/ the base branch, but I might miss something here

thanks !

**Comment by dmitrijus** [comment]
+1

File: DQMOffline/Trigger/plugins/BPHMonitor.cc [hunk 2]
```diff
@@ -13,16 +13,55 @@ BPHMonitor::BPHMonitor( const edm::ParameterSet& iConfig ) :
   folderName_             ( iConfig.getParameter<std::string>("FolderName") )
   , muoToken_             ( mayConsume<reco::MuonCollection>       (iConfig.getParameter<edm::InputTag>("muons")     ) )  
   , bsToken_              ( mayConsume<reco::BeamSpot>             (iConfig.getParameter<edm::InputTag>("beamSpot")))
+  , trToken_              ( mayConsume<reco::TrackCollection>             (iConfig.getParameter<edm::InputTag>("tracks")))
```
Comment (davidlange6) [inline DQMOffline/Trigger/plugins/BPHMonitor.cc]: hi @mtosi - my confusion is that some (the ones in BPHMonitor.cc for example) changes are already part of the master branch. I don't want to figure out what is reviewed and accepted already. 

I'm confused why these are showing up in your PR (even as a commit you made)?

**Comment by mtosi** [comment]
(thanks to @fwyzard )
I --finally-- got which was my issue (sorry !!!!!!)
I was doing the rebase w.r.t. 94x in my local area w/o having updated it to the most recent one !!!!

I fetched the master and rebased,
this branch should be really updated w.r.t. the current master, now

it seems fine to me, now
(at least it seems to reflect what I'm expecting)

please, let me know if it is fine
thanks !!!! (and sorry !!!!!!!!!)

**Comment by mtosi** [comment]
@cmsbuild, please test

**Comment by davidlange6** [comment]
merge

**Pull request merged**